qutebrowser/config/config.py

```diff
@@ -657,6 +666,17 @@ def __init__(self, obj: QObject,
         else:
             self._stylesheet = stylesheet
 
+        if update:
+            if options is None:
+                # try to guess the used config options by parsing the
+                # stylesheet with a simple regex
+                options = frozenset(re.findall(r' conf.([0-9a-zA-Z_.]*) ',
+                                               self._stylesheet))
+
+            # assert that all of the options exists
+            assert all(instance.get_opt(option) for option in options)
```

Comment:
I feel like this can be a bit dangerous, if for example, we have a string value (not evaluated) " conf.foo.bar ", it will cause a crash here. Why not just add it anyway. If we have a false positive that dosen't really matter (only downside is taking up space in the set but I really don't think that's a worry.

Reply:
It implies that there's an error in the code somewhere else. If it's a string value it should pass a custom value for `option` or parse the stylesheet better.

Reply:
It's also possible to use a library to do this (see for example https://stackoverflow.com/a/26319381)